tmva/sofie_parsers/src/RModelParser_ONNX.cxx

```diff
@@ -1175,6 +1175,39 @@ std::unique_ptr<ROperator> make_ROperator_Concat(const onnx::NodeProto &nodeprot
    return op;
 }
 
+   std::unique_ptr<ROperator> make_ROperator_Cast(const onnx::NodeProto& nodeproto, const onnx::GraphProto& /*graphproto */, std::unordered_map<std::string, ETensorType>& tensor_type){
+
+   ETensorType input_type;
+   auto input_name =  nodeproto.input(0);
+   auto it = tensor_type.find(input_name);
+   if (it != tensor_type.end()){
+      input_type = it->second;
+   }else{
+      throw std::runtime_error("TMVA::SOFIE ONNX Parser Cast op has input tensor" + input_name + "  but its type is not yet registered");
+   }
+
+   std::unique_ptr<ROperator> op;
+
```

Comment:
you should check the attribute of the Cast operator to determine the output type. The output type should be stored in the ROperstor_Cast class